src/expr/src/scalar/func.rs

```diff
@@ -818,6 +818,79 @@ fn convert_from<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError>
     }
 }
 
+fn encode<'a>(
+    data: Datum<'a>,
+    format: Datum<'a>,
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    let encoding = format.unwrap_str().to_lowercase();
+    let bytes = data.unwrap_bytes();
+    match encoding.as_str() {
+        "base64" => Ok(Datum::String(
+            temp_storage.push_string(base64::encode(bytes)),
+        )),
+        "hex" => Ok(Datum::String(temp_storage.push_string(hex::encode(bytes)))),
+        "escape" => {
+            let mut result = String::with_capacity(bytes.len());
+            for b in bytes {
+                if b == &('\\' as u8) {
+                    result += "\\\\";
+                } else if (b & 0x80) == 0x80 {
+                    result += &format!("\\{:o}", b);
+                } else {
+                    result += &format!("{}", b);
+                }
+            }
+            Ok(Datum::String(temp_storage.push_string(result)))
+        }
+        _ => Err(EvalError::InvalidEncodingName(encoding)),
+    }
+}
+
+fn decode<'a>(
+    data: Datum<'a>,
+    format: Datum<'a>,
+    temp_storage: &'a RowArena,
+) -> Result<Datum<'a>, EvalError> {
+    let encoding = format.unwrap_str().to_lowercase();
+    let string = data.unwrap_str();
+    match encoding.as_str() {
+        "base64" => match base64::decode(string) {
+            Ok(bytes) => Ok(Datum::from(temp_storage.push_bytes(bytes))),
+            Err(e) => Err(EvalError::InvalidByteSequence {
+                byte_sequence: e.to_string(),
+                encoding_name: encoding,
+            }),
+        },
+        "hex" => match hex::decode(string) {
+            Ok(bytes) => Ok(Datum::from(temp_storage.push_bytes(bytes))),
+            Err(e) => Err(EvalError::InvalidByteSequence {
+                byte_sequence: e.to_string(),
+                encoding_name: encoding,
+            }),
+        },
+        "escape" => {
```

Comment:
The handling of backslashes appears to be different than the one in postgresql:

In postgres:
```
pstoev=# SELECT DECODE('\', 'escape');
ERROR:  invalid input syntax for type bytea
pstoev=# SELECT DECODE('\\', 'escape');
 decode 
--------
 \x5c
(1 row)

pstoev=# SELECT DECODE('\\\', 'escape');
ERROR:  invalid input syntax for type bytea
```

In this branch:

```
materialize=> SELECT DECODE('\', 'escape');
 decode 
--------
 \x5c
(1 row)

materialize=> SELECT DECODE('\\', 'escape');
 decode 
--------
 \x5c5c
(1 row)

materialize=> SELECT DECODE('\\\', 'escape');
  decode  
----------
 \x5c5c5c
(1 row)
```